assignment/final/aLife/code/master.js

```diff
@@ -0,0 +1,417 @@
+    const regex = new RegExp(".*" + pkmnName.value + ".*");
+    let count = pkmnList.filter(pkmn => pkmn.NAME.match(regex)).length;
+    //console.log(pkmnListForTypeFilter);
+    let count2 = pkmnList.length;
+    if(pkmnListForTypeFilter.length == 0){
+        for (let i = 0; i < count; i++) {
+            document.getElementById(pkmnList.filter(pkmn => pkmn.NAME.match(regex))[i].NAME).classList.contains("pkmn") ?
+            document.getElementById(pkmnList.filter(pkmn => pkmn.NAME.match(regex))[i].NAME).classList.remove("pkmn") : "";
+        }
+    }
+    else{
+        try{
+            for(let i = 0; i < pkmnListForTypeFilter[0].LIST.length; i++){
+                document.getElementById(pkmnListForTypeFilter[0].LIST[i].pokemon.name).classList.contains("pkmn")?
+                document.getElementById(pkmnListForTypeFilter[0].LIST[i].pokemon.name).classList.remove("pkmn") : "";
+            }
+        }
+        catch(err){
+
+        }
+    }
+    document.getElementById(x).classList.add("pkmn");
+    pkmnListDiv.classList.remove("pkmn");
+}
+
+function buildNewPkmn(x) {
+    selectNewPkmn(x);
+
+    pkmn = getPkmn(x);
+    builtPkmn.find(i => i.NAME === x) ? "" : builtPkmn.push(pkmn);
+}
+
+function removeTBClicked() {
+    terminalButton.classList.remove("terminalButtonClicked");
+}
+
+function addTBClicked() {
+    terminalButton.classList.add("terminalButtonClicked");
+}
+
+function preBuiltPkmn(pkmn) {
+    //console.log("running in preBuiltPkmn()");
+    currentPkmn = pkmn;
+    selectNewPkmn(pkmn.NAME);
+    moveset.innerHTML = "";
+    terminalScreen.innerHTML =
+        `
+        <img src="${pkmn.SPRITES.front_default}">
+        `;
+    statSection.innerHTML =
+        `
+        <div class="col-sm-6">${pkmn.STATS[0].STAT}: ${pkmn.STATS[0].VALUE}</div>
+        <div class="col-sm-6">${pkmn.STATS[1].STAT}: ${pkmn.STATS[1].VALUE}</div>
+        <div class="col-sm-6">${pkmn.STATS[2].STAT}: ${pkmn.STATS[2].VALUE}</div>
+        <div class="col-sm-6">${pkmn.STATS[3].STAT}: ${pkmn.STATS[3].VALUE}</div>
+        <div class="col-sm-6">${pkmn.STATS[4].STAT}: ${pkmn.STATS[4].VALUE}</div>
+        <div class="col-sm-6">${pkmn.STATS[5].STAT}: ${pkmn.STATS[5].VALUE}</div>
+        `;
+
+    //console.log(pkmn.MOVES);
+    for (let i = 0; i < pkmn.MOVES.length; i++) {
+        moveset.innerHTML +=
+            `
+            <div class="row">
+                <div class="col-sm-5 offset-1">${pkmn.MOVES[i].NAME}</div>
+                <div class="col-sm-2">${pkmn.MOVES[i].TYPE}</div>
+                <div class="col-sm-1" style="text-align: right;">${pkmn.MOVES[i].POWER != null ? pkmn.MOVES[i].POWER : ""}</div>
+            </div>
+            `;
+    }
+
+    idBox.innerHTML = pkmn.ID;
+}
+
+function filterPkmn() {
+    const regex = new RegExp(".*" + pkmnName.value + ".*");
+    let count = pkmnList.filter(pkmn => pkmn.NAME.match(regex)).length;
+    pkmnListDiv.innerHTML = "";
+    for (let i = 0; i < count; i++) {
```

Comment:
if you would of stored the result of your filter:
```
const filteredPokemon = pkmnList.filter(pkmn => pkmn.NAME.match(regex))
```

Then you could call the length any time and not need to store count...as it is really just the `length`.
```
filteredPokemon.forEach( (pokemon, index) => {
  pkmnListDiv.innerHTML += `<p id="${pokemon.NAME}">${pokemon.NAME}</p>`
})
```